web/app/view/Devices.js

```diff
@@ -169,24 +119,46 @@ Ext.define('Traccar.view.Devices', {
             dataIndex: 'contact',
             hidden: true
         }, {
+            text: Strings.groupDialog,
+            dataIndex: 'groupId',
+            hidden: true,
+            filter: {
+                type: 'list',
+                labelField: 'name',
+                store: 'Groups'
+            },
+            renderer: function (value) {
+                var group;
+                if (value !== 0) {
+                    group = Ext.getStore('Groups').getById(value);
+                    return group ? group.get('name') : value;
+                }
+            }
+        }, {
             text: Strings.deviceLastUpdate,
             dataIndex: 'lastUpdate',
-            renderer: function (value, metaData, record) {
-                switch (record.get('status')) {
-                    case 'online':
-                        metaData.tdCls = 'view-color-green';
-                        break;
-                    case 'offline':
-                        metaData.tdCls = 'view-color-red';
-                        break;
-                    default:
-                        metaData.tdCls = 'view-color-yellow';
-                        break;
+            renderer: function (value) {
+                if (value) {
+                    if (Traccar.app.getPreference('twelveHourFormat', false)) {
+                        return Ext.Date.format(value, Traccar.Style.dateTimeFormat12);
+                    } else {
+                        return Ext.Date.format(value, Traccar.Style.dateTimeFormat24);
+                    }
                 }
-                if (Traccar.app.getPreference('twelveHourFormat', false)) {
-                    return Ext.Date.format(value, Traccar.Style.dateTimeFormat12);
-                } else {
-                    return Ext.Date.format(value, Traccar.Style.dateTimeFormat24);
+            }
+        }, {
+            text: Strings.deviceStatus,
+            dataIndex: 'status',
+            flex: 1,
+            filter: {
+                type: 'list',
+                labelField: 'name',
+                store: 'DeviceStatuses'
+            },
+            renderer: function (value, metaData) {
+                if (value) {
```

Comment:
It's turned out that rendering happens twice after device was added. First time with empty string, second time with value from server response (offline).